pkg/envoy/ciliumenvoyconfig_test.go

```diff
@@ -551,6 +555,253 @@ func (s *JSONSuite) TestCiliumEnvoyConfigMulti(c *C) {
+	c.Assert(tc.Hostname, Equals, "host.com:443")
+	c.Assert(tc.UsePost, Equals, true)
+	//
+	// Check cluster resource
+	//
+	c.Assert(cec.Spec.Resources[1].TypeUrl, Equals, "type.googleapis.com/envoy.config.cluster.v3.Cluster")
+	c.Assert(resources.Clusters, HasLen, 1)
+	c.Assert(resources.Clusters[0].Name, Equals, "cluster_0")
+	c.Assert(resources.Clusters[0].ConnectTimeout.Seconds, Equals, int64(5))
+	c.Assert(resources.Clusters[0].ConnectTimeout.Nanos, Equals, int32(0))
+	c.Assert(resources.Clusters[0].LoadAssignment.ClusterName, Equals, "cluster_0")
+	c.Assert(resources.Clusters[0].LoadAssignment.Endpoints, HasLen, 1)
+	c.Assert(resources.Clusters[0].LoadAssignment.Endpoints[0].LbEndpoints, HasLen, 1)
+	addr := resources.Clusters[0].LoadAssignment.Endpoints[0].LbEndpoints[0].GetEndpoint().Address
+	c.Assert(addr, Not(IsNil))
+	c.Assert(addr.GetSocketAddress(), Not(IsNil))
+	c.Assert(addr.GetSocketAddress().GetAddress(), Equals, "127.0.0.1")
+	c.Assert(addr.GetSocketAddress().GetPortValue(), Equals, uint32(10001))
+}
+
+var ciliumEnvoyConfigTCPProxyTermination = `apiVersion: cilium.io/v2
+kind: CiliumEnvoyConfig
+metadata:
+  name: tcp-proxy-ingress-listener
+spec:
+  services:
+  - name: tcp-proxy-ingress
+    namespace: cilium-test
+  resources:
+  - "@type": type.googleapis.com/envoy.config.listener.v3.Listener
+    name: envoy-ingress-listener
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.http_connection_manager
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          stat_prefix: tcp-proxy-ingress-listener
+          route_config:
+            name: local_route
+            virtual_hosts:
+            - name: local_service
+              domains:
+              - "*"
+              routes:
+              - match:
+                  prefix: "/"
+                  headers:
+                  - name: ":method"
+                    string_match:
+                      exact: "POST"
+                route:
+                  cluster: service_google
+                  upgrade_configs:
+                  - upgrade_type: CONNECT
+                    connect_config:
+                      allow_post: true
+          http_filters:
+          - name: envoy.filters.http.router
+          http2_protocol_options:
+            allow_connect: true
+  - "@type": type.googleapis.com/envoy.config.cluster.v3.Cluster
+    name: service_google
+    connect_timeout: 5s
+    type: LOGICAL_DNS
+    # Comment out the following line to test on v6 networks
+    dns_lookup_family: V4_ONLY
+    lb_policy: ROUND_ROBIN
+    load_assignment:
+      cluster_name: service_google
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: www.google.com
+                port_value: 443
+`
+
+func (s *JSONSuite) TestCiliumEnvoyConfigTCPProxyTermination(c *C) {
```

Comment:
Is there a reason we can't just construct a full `Resource` and do a deep comparison instead of a dozen asserts on individual fields?